### PR TITLE
Added partially-stubbed payments journey

### PR DESCRIPTION
**Pull request by stuartmccoll**
## Proposed changes

This pull request adds a partially-stubbed payments journey to the funding-frontend application. This will allow for user research testing of this section of the service.

## Further comments

N/A.

## Screenshot (if applicable)

N/A.

## Checklist

- [X] Added tests for new functionality, or updated existing tests
- [X] Updated `README.md` if necessary

### CODE REVIEW

File: app/controllers/funding_application/gp_project/payment_confirm_details_controller.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+      render :show
+
+    end
+
+  end
+
+  def payments_evidence_params
+
+    # byebug
```
Comment (ptrelease) [inline app/controllers/funding_application/gp_project/payment_confirm_details_controller.rb:36]: remove

File: app/controllers/funding_application/gp_project/payment_confirm_details_controller.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    include FundingApplicationContext, ObjectErrorsLogger
+
+  def update
+
+    logger.info "Updating evidence_file for payment_details ID: " \
+                "#{@funding_application.payment_details.id}"
+
+    @funding_application.payment_details.update(payments_evidence_params)
+
+    # @funding_application.validate_payments_evidence_file = true
```
Comment (ptrelease) [inline app/controllers/funding_application/gp_project/payment_confirm_details_controller.rb:11]: remove

File: app/controllers/funding_application/gp_project/payment_confirm_details_controller.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+    if @funding_application.payment_details.valid?
+
+      logger.info "Finished updating payments_details for ID: " \
+                  "#{@funding_application.payment_details.id}"
+
+      # render :show_confirm_details
```
Comment (ptrelease) [inline app/controllers/funding_application/gp_project/payment_confirm_details_controller.rb:18]: remove

File: app/models/payment_details.rb [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+  validate :encrypt_and_set_account_name, if: :encrypt_account_name?
+
+  KEY = ActiveSupport::KeyGenerator.new(
+    ENV.fetch("PAYMENT_ENCRYPTION_KEY")
+  ).generate_key(
+    ENV.fetch("PAYMENT_ENCRYPTION_SALT"),
+    ActiveSupport::MessageEncryptor.key_len
+  ).freeze
+
+  def encryptor
```
Comment (stuartmccoll) [inline app/models/payment_details.rb:29]: Can we add a docstring to this method?

File: app/models/payment_details.rb [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+  end
+
+  def encrypt_account_name?
+    encrypt_account_name == true
+  end
+
+  # Method to validate that an entered sort code (once stripped of spurious non-numeric characters)
+  # is the correct length. If length is correct, then the sort_code attribute is updated to no
+  # longer contain non-numeric characters, otherwise the errors hash is populated.
+  # If validation passes, turn off validation and write the encrypted value.
```
Comment (stuartmccoll) [inline app/models/payment_details.rb:60]: Might be worth adding something to the final sentence to explain _why_ we turn off validation and write the encrypted value. I'll almost certainly forget by the time we look at this again!

Comment (stuartmccoll) [reply app/models/payment_details.rb:60]: Same thing for the other two methods.

File: app/models/payment_details.rb [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+  def encrypt_and_set_account_name
+    if self.account_name.present?
+      self.encrypt_account_name = false
+      tmp = encryptor.encrypt_and_sign(self.account_name)
+      self.account_name = tmp
+    end
+  end
+
+  def decrypt_account_name
+    if self.account_name.present?
```
Comment (stuartmccoll) [inline app/models/payment_details.rb:110]: Ternary instead?

**Review by ptrelease — APPROVED** [review]

**Pull request merged**